modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/PagedReflectiveOperationExecutorFactory.java

```diff
@@ -49,18 +49,23 @@ public PagedReflectiveOperationExecutorFactory(Class<T> implementationClass, Met
                                                    createDelegate(implementationClass, parameters), pagingProviderTransformer);
   }
 
-  private class PagingProviderResultTransformer implements BiFunction<Object, ExecutionContext<M>, Object> {
+  private class PagingProviderResultTransformer implements CheckedBiFunction<Object, ExecutionContext<M>, Object> {
 
     @Inject
     private ExtensionConnectionSupplier connectionSupplier;
 
     @Override
-    public Object apply(Object result, ExecutionContext<M> executionContext) {
+    public Object applyChecked(Object result, ExecutionContext<M> executionContext) throws Throwable {
       PagingProviderProducer pagingProviderProducer =
           new PagingProviderProducer<>((PagingProvider) result, executionContext.getConfiguration().get(),
                                        (ExecutionContextAdapter) executionContext, connectionSupplier);
       ListConsumer<?> consumer = new ListConsumer(pagingProviderProducer);
-      consumer.loadNextPage();
+      try {
+        consumer.loadNextPage();
+      } catch (Throwable e) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Catch Exception instead of Throwable. ([squid:S1181](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1181))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7717&issues=AWkazdnfotIXnmsUFhS0&open=AWkazdnfotIXnmsUFhS0)

[ISSUE_KEY]: AWkazdnfotIXnmsUFhS0